spring-pulsar/src/main/java/org/springframework/pulsar/core/reactive/ReactivePulsarSenderTemplate.java

```diff
@@ -98,25 +98,30 @@ private Flux<MessageId> doSendMany(String topic, Publisher<T> messages,
 			MessageSpecBuilderCustomizer<T> messageSpecBuilderCustomizer, MessageRouter messageRouter,
 			ReactiveMessageSenderBuilderCustomizer<T> customizer) {
 		final String topicName = ReactiveMessageSenderUtils.resolveTopicName(topic, this.reactiveMessageSenderFactory);
-		this.logger.trace(() -> String.format("Sending reactive msg to '%s' topic", topicName));
+		this.logger.trace(() -> String.format("Sending reactive messages to '%s' topic", topicName));
 
 		if (this.schema != null) {
-			ReactiveMessageSender<T> sender = this.reactiveMessageSenderFactory.createReactiveMessageSender(topic,
-					this.schema, messageRouter,
-					customizer == null ? Collections.emptyList() : Collections.singletonList(customizer));
-
-			Flux<MessageSpec<T>> map = Flux.from(messages)
-					.map(message -> getMessageSpec(messageSpecBuilderCustomizer, message));
-			return sender.sendMessages(map).doOnError(
-					ex -> this.logger.error(ex, () -> String.format("Failed to send msg to '%s' topic", topicName)))
-					.doOnNext(msgId -> this.logger.trace(() -> String.format("Sent msg to '%s' topic", topicName)));
+			/*
+			 * If the template has a schema, we can create the message sender right away
+			 * and use ReactiveMessageSender::sendMessages to send them as a stream.
+			 * Otherwise we need to wait to get a message to create it and we can't share
+			 * it between messages. So we create one each time and use
+			 * ReactiveMessageSender::sendMessage to send messages individually.
+			 */
+			ReactiveMessageSender<T> sender = createMessageSender(topic, null, messageRouter, customizer);
+			return Flux.from(messages).map(message -> getMessageSpec(messageSpecBuilderCustomizer, message))
+					.as(sender::sendMessages)
+					.doOnError(ex -> this.logger.error(ex,
+							() -> String.format("Failed to send messages to '%s' topic", topicName)))
+					.doOnNext(
+							msgId -> this.logger.trace(() -> String.format("Sent messages to '%s' topic", topicName)));
```

Comment:
Hehe. This particular msg was correct before - as it was a single msg in `onNext`... We can change it later though.